test/functional/wallet_backwards_compatibility.py

```diff
@@ -131,6 +131,60 @@ def test_v19_addmultisigaddress(self):
         wallet = node_v19.get_wallet_rpc("w1_v19")
         assert wallet.getaddressinfo(address_18075)["solvable"]
 
+    def test_downgrade_encryption(self, node_master, descriptors_nodes, legacy_nodes):
+        self.log.info("Test downgrade encryption + re-open on master")
+
+        # Create wallet on master, downgrade and encrypt the wallet. Then, try to open it again on master.
+        name_wallet_on_master = "new_wallet"
+        node_master.createwallet(name_wallet_on_master, descriptors=self.options.descriptors)
+        wallet_on_master = node_master.get_wallet_rpc(name_wallet_on_master)
+
+        # Make a backup of the wallet file
+        backup_path = os.path.join(self.options.tmpdir, f"{name_wallet_on_master}.dat")
+        wallet_on_master.backupwallet(backup_path)
+
+        # Downgrade and encrypt the wallet.
+        for node in descriptors_nodes if self.options.descriptors else legacy_nodes:
+            if self.major_version_less_than(node, 22):
+                continue  # cannot downgrade further than v22, tr() descriptors aren't supported.
```

Comment:
You could create a pre-v22 wallet, upgrade to master (which won't add the `tr()` descriptor) and then continue the test.

Although I don't think we care that deeply about downgrading that far, testing a wallet without a `tr()` descriptor is useful - for followup PR's that enable adding that desciptor.